pkg/injector/init_container.go

```diff
@@ -12,6 +12,8 @@ func getInitContainerSpec(containerName string, cfg configurator.Configurator, o
 	inboundPortExclusionList []int, enablePrivilegedInitContainer bool) corev1.Container {
 	iptablesInitCommandsList := generateIptablesCommands(outboundIPRangeExclusionList, outboundPortExclusionList, inboundPortExclusionList)
 	iptablesInitCommand := strings.Join(iptablesInitCommandsList, " && ")
+	runInitContainerAsNonRoot := false
```

Comment:
The init container should not be run as `root` by default. This should be made configurable based on whether the backing pod has a security context that prevents non-root containers from executing iptables.

I would expect the init container to not run as `root` by default, which means `SecurityContext.RunAsNonRoot = true`.

Reply:
By default, with no security contexts, the init container runs as root. However, the pod level security context can override this. The init container must run as root to program iptables. This has been tested by setting the init container SecurityContext to runAsNonRoot and run with a non zero user id. This results in the init container failing.

The init container cannot program iptables if it is non-root. This is unaffected by pod-level security contexts.

Reply:
Have validated the above by setting `runAsNonRoot` to true and the user ID to 1000 with our default bookstore demo (no additional security contexts) - this causes the same xtables permission denied error we are trying to resolve. If we were to set it to true and keep the user ID as 0, this will be in opposition to what the [k8s API docs](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#securitycontext-v1-core) say this value checks for:
> If true, the Kubelet will validate the image at runtime to ensure that it does not run as UID 0 (root) and fail to start the container if it does. If unset or false, no such validation will be performed.